### PR TITLE
Hellfire Watchers should cast heal (30643) when Watchkeeper Gargolmar…

### PR DESCRIPTION
**Pull request by miraco**
… hits 40%

## 🍰 Pullrequest

Let him SendAIEvent (AI_EVENT_CUSTOM_EVENTAI_A) to trigger Hellfire Watcher's EventAI 
https://github.com/cmangos/tbc-db/pull/1152

### Proof

Tested on Classic WOTLK

### Issues

- None

### How2Test

- None

### Todo / Checklist

- [X] None

### CODE REVIEW

File: src/game/AI/ScriptDevAI/scripts/outland/hellfire_citadel/hellfire_ramparts/boss_watchkeeper_gargolmar.cpp [hunk 1]
```diff
@@ -136,6 +136,15 @@ struct boss_watchkeeper_gargolmarAI : public CombatAI
                 if (m_creature->GetHealthPercent() < 40.0f)
                 {
                     DoScriptText(SAY_HEAL, m_creature);
+
+                    CreatureList WatcherList;
```
Comment (killerwife) [inline src/game/AI/ScriptDevAI/scripts/outland/hellfire_citadel/hellfire_ramparts/boss_watchkeeper_gargolmar.cpp:140]: So I will be a dickbutt, local variables, camelCase not PascalCase.

File: src/game/AI/ScriptDevAI/scripts/outland/hellfire_citadel/hellfire_ramparts/boss_watchkeeper_gargolmar.cpp [hunk 2]
```diff
@@ -136,6 +136,15 @@ struct boss_watchkeeper_gargolmarAI : public CombatAI
...
+                    GetCreatureListWithEntryInGrid(WatcherList, m_creature, NPC_HELLFIRE_WATCHER, 100.0f);
+                    for (auto& itr : WatcherList)
```
Comment (killerwife) [inline src/game/AI/ScriptDevAI/scripts/outland/hellfire_citadel/hellfire_ramparts/boss_watchkeeper_gargolmar.cpp:142]: Itr implies its an iterator. This is very much a creature, so if possible, name it that way. `Creature* watcher :` is waaay better to read.

Comment (miraco) [reply src/game/AI/ScriptDevAI/scripts/outland/hellfire_citadel/hellfire_ramparts/boss_watchkeeper_gargolmar.cpp]: I mostly copy pasted it by another script, should be solved now

File: src/game/AI/ScriptDevAI/scripts/outland/hellfire_citadel/hellfire_ramparts/boss_watchkeeper_gargolmar.cpp [hunk 3]
```diff
@@ -136,6 +136,15 @@ struct boss_watchkeeper_gargolmarAI : public CombatAI
                 if (m_creature->GetHealthPercent() < 40.0f)
                 {
                     DoScriptText(SAY_HEAL, m_creature);
+
+                    CreatureList watcherList;
+                    GetCreatureListWithEntryInGrid(watcherList, m_creature, NPC_HELLFIRE_WATCHER, 100.0f);
+                    for (Creature* watcher: watcherList)
```
Comment (killerwife) [inline src/game/AI/ScriptDevAI/scripts/outland/hellfire_citadel/hellfire_ramparts/boss_watchkeeper_gargolmar.cpp:142]: Now you are missing a space :D But in spirit of cooperation I will fix that myself.

**Pull request closed without merging**